operations/mimir-mixin/alerts/ingest-storage.libsonnet

```diff
@@ -19,6 +19,110 @@
             message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is failing to commit the last consumed offset.' % $._config,
           },
         },
+
+        {
+          alert: $.alertName('IngesterFailedToReadRecordsFromKafka'),
+          'for': '5m',
+
+          // Metric used by this alert is reported by Kafka client on read errors from connection to Kafka.
+          // We use node_id to only alert if problems to the same Kafka node are repeating.
+          // If problems are for different nodes (eg. during rollout), that is not a problem, and we don't need to trigger alert.
+          expr: |||
+            sum by(%(alert_aggregation_labels)s, %(per_instance_label)s, node_id) (rate(cortex_ingest_storage_reader_read_errors_total[1m]))
+            > 0
+          ||| % $._config,
+          labels: {
+            severity: 'critical',
+          },
+          annotations: {
+            message: '%(product)s {{ $labels.%(per_instance_label)s }} in %(alert_aggregation_variables)s is failing to read records from Kafka.' % $._config,
+          },
+        },
+
+        {
+          alert: $.alertName('IngesterKafkaFetchErrorsRateTooHigh'),
+          'for': '15m',
+          // See https://github.com/grafana/mimir/blob/24591ae56cd7d6ef24a7cc1541a41405676773f4/vendor/github.com/twmb/franz-go/pkg/kgo/record_and_fetch.go#L332-L366 for errors that can be reported here.
+          expr: |||
+            sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (rate (cortex_ingest_storage_reader_fetch_errors_total[5m]))
+            /
+            sum by (%(alert_aggregation_labels)s, %(per_instance_label)s) (rate (cortex_ingest_storage_reader_fetches_total[5m]))
+            > 0.1
+          ||| % $._config,
+          labels: {
+            severity: 'critical',
```

Comment:
wouldn't we get alerted for increased lag if there are errors? 10% of errors might be tolerable 🤷 I'm only trying to reduce the number of alerts so we don't get alert fatigue from the start

Reply:
Here the condition must be true for 15 minutes. I don't think 10% errors during that time is tolerable -- normally we expect 0 errors.

Reply:
> wouldn't we get alerted for increased lag if there are errors?

I client is able to read further records, then this wouldn't be true.

Reply:
Isn't this alerting on symptoms vs causes? Errors themselves are causes of outages, but they aren't symptoms of an outage. Increased delay or failing consistency checks are the symptoms.

We might not have to immediately address the errors if all the customer-visible metrics look ok (like delay and strong consistency guarantees)

Don't want to delve too deep here, but I suspect this alert will trigger along with some other alert and won't bring much value on its own. If you think we should keep it, then I can also disagree and commit 😄

Reply:
Honestly I don't know yet. I guess first few incidents will tell us more.